src/highlight/HighlightPromoter.ts

```diff
@@ -0,0 +1,92 @@
+import debounce from 'lodash/debounce';
+import { AppStore, getIsInitialized, SelectionArg, setSelectionAction } from '../store';
+import { MOUSE_PRIMARY } from '../constants';
+
+export type Options = {
+    getSelection: () => SelectionArg | null;
+    selectionChangeDelay?: number;
+    store: AppStore;
+};
+
+// Debounce 500ms for keyboard selection
+const SELECTION_CHANGE_DEBOUNCE = 500;
+
+export default class HighlightPromoter {
+    annotatedEl?: HTMLElement;
+
+    getSelection: () => SelectionArg | null;
+
+    isMouseSelecting = false;
+
+    selectionChangeDelay = 0;
+
+    selectionChangeTimer?: number;
+
+    store: AppStore;
+
+    constructor({ getSelection, selectionChangeDelay = 0, store }: Options) {
```

Comment:
Would it make sense to default `selectionChangeDelay` to the 300ms magic number?

Reply:
I think 300ms magic number is only related to pdfjs, which is only related to DocumentAnnotator. For the other annotators which don't use pdfjs, it should be 0.